pkgs/tools/typesetting/tectonic/default.nix

```diff
@@ -23,8 +32,20 @@ rustPlatform.buildRustPackage rec {
     sha256 = "sha256-Cd8YzjU5mCA5DmgLBjg8eVRc87chVVIXinJuf8cNw3o=";
   };
 
+  patches = [
+    # Provides a version lock of the tectonic web bundle for reproducible builds
+    # by specifying the environment variable `TECTONIC_WEB_BUNDLE_LOCKED`.
+    # Upstream PR: https://github.com/tectonic-typesetting/tectonic/pull/1131
+    # This patch should be removed once the upstream PR is merged
+    (fetchpatch {
```

Comment:
What is the point in fetching the patch if you don't inject the `$TECTONIC_WEB_BUNDLE_LOCKED` variable? I'm a rather against this patch, but even if I do accept it, It doesn't make sense to me to add it here unconditionally, vs only in the main `tectonic` derivation. Note also that you anyway need to `overrideAttrs` since you rely upon build time environment variable.